tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -97,20 +97,37 @@ func (s *stresser) Stress() error {
 func (s *stresser) run(ctx context.Context, kvc pb.KVClient) {
 	defer s.wg.Done()
 
-	for {
-		if err := s.rateLimiter.Wait(ctx); err == context.Canceled {
+	var (
+		key   []byte
+		value []byte
+	)
+	for i := 0; ; i++ {
+		err := s.rateLimiter.Wait(ctx)
+		if err == context.Canceled {
 			return
 		}
 
 		// TODO: 10-second is enough timeout to cover leader failure
 		// and immediate leader election. Find out what other cases this
 		// could be timed out.
-		putctx, putcancel := context.WithTimeout(ctx, 10*time.Second)
-		_, err := kvc.Put(putctx, &pb.PutRequest{
-			Key:   []byte(fmt.Sprintf("foo%d", rand.Intn(s.KeySuffixRange))),
-			Value: []byte(randStr(s.KeySize)),
-		})
-		putcancel()
+		sctx, scancel := context.WithTimeout(ctx, 10*time.Second)
+
+		idx := i % 5
```

Comment:
can this be done be done a bit more generically? something like:
```
type stressEntry struct {p float32; f stressFunc }
type stressTable []stressEntry
func (st *stressTable) choose() stressFunc
```

I take it the index isn't really important since the stresser is non-deterministic because of interleaving anyway?